docs/examples/extending-1-classes/index.md

```diff
@@ -5,200 +5,175 @@ title: Extending Leaflet, Class Theory
+
+
+```js
+class RotateMarker extends Marker {
+	initialize(latlng, rotation, options) {
+		super.initialize(latlng, options);
+		this._rotation = rotation;
+	}
+}
+```
+
+### Naming conventions
 
+When creating Leaflet classes, adhere to these conventions:
 
+* Class names should be in `UpperCamelCase`.
+* Method and property names should be in `lowerCamelCase`.
+* Private properties and methods start with an underscore (`_`). This indicates they're internal and shouldn't be used directly.
+
+### Setting default options
+
+All classes that extend from `L.Class` can be provided with default options by calling `setDefaultOptions()` in a [static initialization block](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Classes/Static_initialization_blocks):
+
+```js
+class MyBox extends Class {
+	static {
+		this.setDefaultOptions({
+			width: 1,
+			height: 1
+		});
+	}
+
+	initialize(name, options) {
+		super.initialize(options);
+		this.name = name;
+	}
+}
+
+const instance = new MyBox('Red', {width: 10});
+
+console.log(instance.name); // Outputs "Red"
+console.log(instance.options.width); // Outputs "10"
+console.log(instance.options.height); // Outputs "1", the default
+```
+
+These options are inherited from parent classes, and merged automatically:
+
+```js
+class MyCube extends MyBox {
+	static {
+		this.setDefaultOptions({
+			depth: 1
+		});
+	}
+}
+
+const cube = new MyCube('Blue');
+
+console.log(cube.options.width); // Outputs "1", parent class default
+console.log(cube.options.height); // Outputs "1", parent class default
+console.log(cube.options.depth); // Outputs "1"
+```
+
+### `L.Class.include()`
+
+Leaflet provides `.include()` to add or override methods in existing classes. This is useful for monkey-patching or adding functionality to base classes:
+
+```js
+class MyLayer extends Layer {
+	initialize(options) {
+		super.initialize(options);
+		this._count = 0;
+	}
+
+	incrementCount() {
+		return ++this._count;
+	}
+}
+
+// Add new methods or override existing ones
+MyLayer.include({
+	_getCountStep() {
+		return 2;
+	},
+
+	incrementCount() {
+		return this._count += this._getCountStep();
+	}
+});
+
+const instance = new MyLayer();
+console.log(instance.incrementCount()); // Outputs "2"
+```
+
+Note: Use `.include()` sparingly, as modifying base classes can have unexpected side effects. As a general rule of thumb, try to extend existing classes instead.
+
+### Initialization hooks
```

Comment:
I'm wondering if we should rephrase this whole section, and mark "init hooks" as legacy.

Nowadays I'd expect any initialization to happen in the constructor, and the developer is supposed to call `super.constructor()`.

Reply:
How would you solve, for example, extending the Marker initialization without creating a new class? Many plugins rely on this logic, and it is very useful for cross-plugin interactions.